javascripts/controllers/NewCtrl.js

```diff
@@ -1,5 +1,21 @@
 'use strict';
 
-app.controller("NewCtrl", function($scope) {
-    $scope.controller = "NewCtrl";
+app.controller("NewCtrl", function($http, $location, $rootScope, $scope, DatabaseService) {
+   $scope.newcontact = {};
+   
+   $scope.submitForm = () => {
+        let newContact = {
+            "first_name": $scope.newcontact.firstname,
+			"last_name": $scope.newcontact.lastname,
+			"phone_number": $scope.newcontact.phonenumber,
+			"email": $scope.newcontact.email,
+			"twitter": $scope.newcontact.twitter,
+			"facebook_page": $scope.newcontact.facebook,
+            "instagram_username": $scope.newcontact.instagram,
+            "user_id": $rootScope.uid
+        };
+
+        DatabaseService.addNewContact(newContact);
```

Comment:
Because you aren't using .then() here, the contacts aren't loading but the time it switches location.

Reply:
@morecallan updated!